tss-esapi/src/context/handle_manager.rs

```diff
@@ -41,8 +41,12 @@ impl HandleManager {
         }
 
         if self.open_handles.contains_key(&handle) {
-            error!("Handle({}) is already open", ESYS_TR::from(handle));
-            return Err(Error::local_error(WrapperErrorKind::InvalidHandleState));
+            // It is safe to call unwrap because the existance of the key has already been verified.
+            let stored_handle_drop_action = self.open_handles.get(&handle).unwrap();
+            if handle_drop_action != *stored_handle_drop_action {
+                error!("Handle drop action inconsistency");
+                return Err(Error::local_error(WrapperErrorKind::InconsistentParams));
+            }
         }
         let _ = self.open_handles.insert(handle, handle_drop_action);
```

Comment:
:thinking: is `let _` necessary? It seems `insert` returns an `Option` and it's not marked with `#[must_use]`... ah... this is outside of this change... sorry :)